public/app/features/canvas/elements/cloud.tsx

```diff
@@ -0,0 +1,175 @@
+      },
+    },
+    config: {
+      align: Align.Center,
+      valign: VAlign.Middle,
+      color: {
+        fixed: defaultTextColor,
+      },
+    },
+    placement: {
+      width: options?.placement?.width ?? 110,
+      height: options?.placement?.height ?? 70,
+      top: options?.placement?.top,
+      left: options?.placement?.left,
+    },
+  }),
+
+  // Called when data changes
+  prepareData: (dimensionContext: DimensionContext, elementOptions: CanvasElementOptions<CanvasElementConfig>) => {
+    const textConfig = elementOptions.config;
+
+    const data: CanvasElementData = {
+      text: textConfig?.text ? dimensionContext.getText(textConfig.text).value() : '',
+      align: textConfig?.align ?? Align.Center,
+      valign: textConfig?.valign ?? VAlign.Middle,
+      size: textConfig?.size,
+    };
+
+    if (textConfig?.color) {
+      data.color = dimensionContext.getColor(textConfig.color).value();
+    }
+
+    data.links = getDataLinks(dimensionContext, elementOptions, data.text);
+
+    const { background, border } = elementOptions;
+    data.backgroundColor = background?.color ? dimensionContext.getColor(background.color).value() : defaultBgColor;
+    data.borderColor = border?.color ? dimensionContext.getColor(border.color).value() : defaultBgColor;
+    data.borderWidth = border?.width ?? 0;
+
+    return data;
+  },
+
+  registerOptionsUI: (builder) => {
+    const category = ['Cloud'];
+    builder
+      .addCustomEditor({
+        category,
+        id: 'textSelector',
+        path: 'config.text',
+        name: 'Text',
+        editor: TextDimensionEditor,
+      })
+      .addCustomEditor({
+        category,
+        id: 'config.color',
+        path: 'config.color',
+        name: 'Text color',
+        editor: ColorDimensionEditor,
+        settings: {},
+        defaultValue: {},
+      })
+      .addRadio({
+        category,
+        path: 'config.align',
+        name: 'Align text',
+        settings: {
+          options: [
+            { value: Align.Left, label: 'Left' },
+            { value: Align.Center, label: 'Center' },
+            { value: Align.Right, label: 'Right' },
+          ],
+        },
+        defaultValue: Align.Left,
+      })
+      .addRadio({
+        category,
+        path: 'config.valign',
+        name: 'Vertical align',
+        settings: {
+          options: [
+            { value: VAlign.Top, label: 'Top' },
+            { value: VAlign.Middle, label: 'Middle' },
+            { value: VAlign.Bottom, label: 'Bottom' },
+          ],
+        },
+        defaultValue: VAlign.Middle,
+      })
+      .addNumberInput({
+        category,
+        path: 'config.size',
+        name: 'Text size',
+        settings: {
+          placeholder: 'Auto',
+        },
+      });
+  },
+};
+
+const getStyles = (theme: GrafanaTheme2, data: CanvasElementData | undefined) => ({
+  container: css({
+    height: '100%',
+    width: '100%',
+  }),
+  text: css({
+    position: 'absolute',
```

Comment:
Computing these booleans outside of the properties would make this more readable. Especially since there's only a handful of values